src/test/scala-2.11/kamon/testkit/BaseKamonSpec.scala

```diff
@@ -0,0 +1,66 @@
+/*
+ * =========================================================================================
+ * Copyright © 2013-2015 the kamon project <http://kamon.io/>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ * =========================================================================================
+ */
+
+package kamon.testkit
+
+import akka.actor.ActorSystem
+import akka.testkit.{ ImplicitSender, TestKitBase }
+import com.typesafe.config.Config
+import kamon.Kamon
+import kamon.metric.{ Entity, EntitySnapshot, SubscriptionsDispatcher }
+import kamon.trace.TraceContext
+import kamon.util.LazyActorRef
+import org.scalatest.{ BeforeAndAfterAll, Matchers, WordSpecLike }
+
+abstract class BaseKamonSpec(actorSystemName: String) extends TestKitBase with WordSpecLike with Matchers with ImplicitSender with BeforeAndAfterAll {
```

Comment:
The need to split between 2.11 and 2.12 appears to be an artifact of how `class` files are merged into the resulting output.   SBT will aggregate the two source roots `/src/test/scala` and `/src/test/scala-2.1x`. Thus, when compiling the resulting output complains (and halts) informing us that there are two definitions of the type.

You can see this through out the changes I've proposed where support for 2.10 was still desired.